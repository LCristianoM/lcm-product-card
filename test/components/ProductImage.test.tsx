
import React from 'react';
import { ProductImage } from '../../src/components/ProductImage';
import TestRenderer from 'react-test-renderer';
import { product2 } from '../data/products';
import { ProductCard  } from '../../src/components/ProductCard';

describe('ProductImage', () => {
  test('debe mostrar el componete correctamene con la imagen', () => {

    const wrapper = TestRenderer.create(<ProductImage img="https://hola.png" />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});

test('debe mostrar el componete con la imagen del producto', () => {
  const wrapper = TestRenderer.create(
    <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
  );
  expect(wrapper.toJSON()).toMatchSnapshot();
});
