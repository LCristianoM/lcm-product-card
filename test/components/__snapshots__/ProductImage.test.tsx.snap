// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe mostrar el componete correctamene con la imagen 1`] = `
<img
  alt="Product Img"
  className="productImg undefined"
  src="https://hola.png"
/>
`;

exports[`debe mostrar el componete con el nombre del producto 1`] = `
<div
  className="productCard undefined "
>
  <img
    alt="Product Img"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`debe mostrar el componete con la imagen del producto 1`] = `
<div
  className="productCard undefined "
>
  <img
    alt="Product Img"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
